rest_framework_json_api/views.py

```diff
@@ -98,6 +100,54 @@ def get_queryset(self, *args, **kwargs):
         return qs
 
 
+class RelatedMixin(object):
+    """
+    This mixin handles all related entities, whose Serializers are declared in "related_serializers"
+    """
+    related_serializers = {}
+    field_name_mapping = {}
+
+    def retrieve_related(self, request, *args, **kwargs):
+        serializer_kwargs = {}
+        instance = self.get_related_instance()
+
+        if hasattr(instance, 'all'):
+            instance = instance.all()
+
+        if callable(instance):
+            instance = instance()
+
+        if instance is None:
+            return Response(data=None)
+
+        if isinstance(instance, Iterable):
+            serializer_kwargs['many'] = True
+
+        serializer = self.get_serializer(instance, **serializer_kwargs)
+        return Response(serializer.data)
+
+    def get_serializer_class(self):
+        if 'related_field' in self.kwargs:
+            field_name = self.get_related_field_name()
+            _class = self.related_serializers.get(field_name, None)
+            if _class is None:
+                raise NotFound
+            return _class
+        return super(RelatedMixin, self).get_serializer_class()
+
+    def get_related_field_name(self):
+        field_name = self.kwargs['related_field']
+        if field_name in self.field_name_mapping:
+            return self.field_name_mapping[field_name]
+        return field_name
+
+    def get_related_instance(self):
+        try:
+            return getattr(self.get_object(), self.get_related_field_name())
+        except AttributeError:
+            raise NotFound
```

Comment:
same here as above.

Reply:
If there is no attribute on model - return 404. This is similar behavior to the answer above

Reply:
that would be more user friendly and a good improvement.

Reply:
Just tried to do that. But what if we want to force api/author/1/bio/ return 404 ? It looks like impossible, because it will start giving 500 all the time. So, I think we have to leave it as is.

Reply:
I guess it could be by looking up whether the related field is defined on the serializer. If yes then it would be a 500 error if it is not defined or if it is defined a 404.

However I see this might be a bit of a overkill just for an error message. Let's leave it as it is and make sure that it is well documented.

Reply:
>I guess it could be by looking up whether the related field is defined on the serializer. If yes then it would be a 500 error if it is not defined or if it is defined a 404.
Not sure I got you :) So, if there is no attribute on a model, we check if the attribute is declared on a serializer. If it is declared - we return 500 saying smth like "Wrong config" and if it is not declared - return 404 ? 

I will agree with case return 404, because in "500" case we should just get correct model attribute name from related field "source".

Well, honestly speaking I don't like the fact that RelatedMixin depends on parent's entity serializer class, like fetching field source name and so on. Probably that is unreasonable, but who knows, haha. I like keep thing as simple as possible, then its easier to support it.

So, I would keep it as is. Any minds ?

Reply:
As said above I also think this is a bit of a overkill. It was just some thoughts on how it could be done.

But let's leave it as is and let's make sure the documentation is clear on it.